src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -718,26 +718,36 @@ class Simpletask : ThemedNoActionBarActivity() {
     }
 
     private fun deleteTasks(tasks: List<TodoItem>) {
-        showConfirmationDialog(this, R.string.delete_task_message, DialogInterface.OnClickListener { dialogInterface, i ->
+        val numTasks = tasks.size
+        var title = getString(R.string.delete_task_title)
+                    .replaceFirst(Regex("%s"), numTasks.toString())
+        val delete = DialogInterface.OnClickListener { dialogInterface, i ->
             for (t in tasks) {
                 TodoList.remove(t)
             }
             TodoList.notifyChanged(Config.todoFileName, Config.eol, m_app, true)
             invalidateOptionsMenu()
-        }, R.string.delete_task_title)
-    }
+        }
 
-    private fun archiveTasks(tasksToArchive: List<TodoItem>?, areYouSureDialog: Boolean) {
```

Comment:
We were using a hack where if the list was null (ie, no selection), we archive all tasks instead. That was ugly, so let's not make this nullable anymore.